packages/repository-tests/src/crud/relations/helpers.ts

```diff
@@ -0,0 +1,47 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/repository-tests
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {juggler} from '@loopback/repository';
+import {CrudRepositoryCtor} from '../..';
+import {
+  createOrderRepo,
+  createAddressRepo,
+  createCustomerRepo,
+  createShipmentRepo,
+} from './fixtures/repositories';
+
+export function givenBoundCrudRepositories(
+  db: juggler.DataSource,
+  repositoryClass: CrudRepositoryCtor,
+) {
+  // get the repository class and create a new instance of it
+  const customerRepoClass = createCustomerRepo(repositoryClass);
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const customerRepo: any = new customerRepoClass(
+    db,
+    async () => orderRepo,
+    async () => addressRepo,
+  );
+
+  const orderRepoClass = createOrderRepo(repositoryClass);
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const orderRepo: any = new orderRepoClass(
+    db,
+    async () => customerRepo,
+    async () => shipmentRepo,
+  );
+
+  const shipmentRepoClass = createShipmentRepo(repositoryClass);
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const shipmentRepo: any = new shipmentRepoClass(db, async () => orderRepo);
+
+  const addressRepoClass = createAddressRepo(repositoryClass);
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const addressRepo: any = new addressRepoClass(db, async () => customerRepo);
+
+  return {customerRepo, orderRepo, shipmentRepo, addressRepo};
+}
+
+export type mixedIdType = string | number;
```

Comment:
We prefer to use PascalCase for type names.

```suggestion
export type MixedIdType = string | number;
```

Reply:
Please add a tsdoc comment explaining the reasons for introducing this type. I think this type should be used by all CRUD tests, not only those specific to relations. Let's move this type definition to `packages/repository-tests/src/helpers.repository-tests.ts`.

Reply:
On the second thought, do we really care about the actual type, considering that it's a union `number | string` and therefore we can do very little with those values? What if we used `unknown` everywhere we need to describe the type of a PK or a FK property?